src/main/java/org/gridsuite/cgmes/boundary/server/CgmesBoundaryService.java

```diff
@@ -38,7 +42,23 @@ public CgmesBoundaryService(BoundaryRepository boundaryRepository) {
 
     Optional<BoundaryInfo> getBoundary(String boundaryId) {
         Optional<BoundaryEntity> boundary = boundaryRepository.findById(boundaryId);
-        return boundary.map(b -> new BoundaryInfo(b.getId(), b.getFilename(), new String(b.getBoundary().array(), StandardCharsets.UTF_8)));
+        return boundary.map(b -> new BoundaryInfo(b.getId(), b.getFilename(), b.getScenarioTime(), new String(b.getBoundary().array(), StandardCharsets.UTF_8)));
+    }
+
+    BoundaryInfo getLastBoundary(String profile) {
+        List<BoundaryInfo> boundaries = getBoundariesList();
```

Comment:
This is exactly why we should port these kind of micro service to relational db, it would be easy to get last boundary by just a SQL query. Here is not a clean approach to get everything and filter, but I understand that it would need to denormalize data in the DB and would be painful.